src/components/SimpleDatePicker/index.tsx

```diff
@@ -2,49 +2,93 @@ import { DayPicker, WeekdayProps } from "react-day-picker";
+import { useEffect, useState } from "react";
 import dayjs from "dayjs";
+import clsx from "clsx";
 interface ISimpleDatePickerProps {
-  isVisible?: boolean;
   mode?: "single" | "multiple" | "range";
+  value?: string;
+  defaultVulue?: string;
+  className?: string;
+  onChange?(t: string): void;
   onChange?: (value: string) => void;
 }
 
 const SimpleDatePicker = (props: ISimpleDatePickerProps) => {
-  const { isVisible, onChange, ...dayPickerProps } = props;
-  const [selected, setSelected] = useState<Date>();
+  const { value, defaultVulue, className, onChange, ...dayPickerProps } = props;
+  const [isVisible, setIsVisible] = useState(false);
+  const [selected, setSelected] = useState<string>(
+    defaultVulue && dayjs(defaultVulue || "").isValid() ? defaultVulue : ""
+  );
+
+  const formatDate = (dateInput: string) => {
+    const date = dayjs(dateInput);
+    const currentYear = dayjs().year();
+
+    if (date.year() === currentYear) {
+      return date.format("DD MMM");
+    } else {
+      return date.format("DD MMM YYYY");
+    }
+  };
 
   const handleConfirm = () => {
-    onChange?.(selected ? dayjs(selected).format("YYYY-MM-DD") : "");
+    const selectedDate = selected ? dayjs(selected).format("YYYY-MM-DD") : "";
+    onChange?.(selectedDate);
+    setIsVisible(false);
   };
 
+  useEffect(() => {
+    if (value && dayjs(value).isValid()) {
+      setSelected(value);
+    }
+  }, [value]);
+
   return (
-    <Drawer
-      isVisible={isVisible}
-      direction="bottom"
-      rootClassName="px-[17.5px] pt-5 pb-7 bg-tertiary"
-    >
-      <DayPicker
-        {...dayPickerProps}
-        mode="single"
-        selected={selected}
-        onSelect={setSelected}
-        captionLayout="dropdown"
-        weekStartsOn={1}
-        components={{
-          Weekday: (props: WeekdayProps) => {
-            return <th>{props['aria-label']?.slice(0, 3)}</th>;
-          }
-        }}
-        className="simple-date-picker"
-      />
-      <button
-        className="w-full mt-2 mx-[2.5px] bg-primary rounded-[24px] text-[14px] font-bold py-[10px] font-outfit leading-[25px]"
-        onClick={handleConfirm}
+    <>
+      <div
+        className={clsx(
+          "relative py-[12px] pl-[14px] pr-[40px] border border-tertiary rounded-[10px]",
+          className
+        )}
+        onClick={() => setIsVisible(true)}
+      >
+        <span className="block min-w-[50px] h-[20px] font-normal text-[14px] text-input-placeholder leading-[20px]">
+          {selected && formatDate(selected)}
+        </span>
+
+        <i className="absolute top-1/2 right-[14px] -translate-y-1/2 votigram-icon-navbar-vote text-input-placeholder text-[18px]" />
+      </div>
+      <Drawer
+        isVisible={isVisible}
+        direction="bottom"
+        onClose={setIsVisible}
+        rootClassName="px-[17.5px] pt-5 pb-7 bg-tertiary"
       >
-        Confirm
-      </button>
-    </Drawer>
+        <DayPicker
+          {...dayPickerProps}
```

Comment:
spread operator should always put it at the end so developer can pass in and overwrite default value

Reply:
Updated